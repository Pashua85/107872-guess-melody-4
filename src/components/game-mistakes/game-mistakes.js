import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {getMistakes} from '../../store/reducers/gameReducer/selectors';

const GameMistakes = (props) => {
  const wrongs = [];
  for (let i = 0; i < props.mistakes; i++) {
    wrongs.push(
        <div className="wrong" key={i}></div>
    );
  }
  return (
    <div className="game__mistakes">
      {wrongs}
    </div>
  );
};

GameMistakes.propTypes = {
  mistakes: PropTypes.number.isRequired
};

const mapStateToProps = (state) => ({
  mistakes: getMistakes(state)
});

export default connect(mapStateToProps)(GameMistakes);
export {GameMistakes};
