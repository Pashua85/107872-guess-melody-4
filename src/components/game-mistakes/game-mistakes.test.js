import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {GameMistakes} from './game-mistakes';

Enzyme.configure({
  adapter: new Adapter()
});

describe(`GameMistkes`, () => {
  it(`should render correctly with no mistakes`, () => {
    const wrapper = shallow(<GameMistakes mistakes={0} />);
    expect(wrapper).toMatchSnapshot();
  });

  it(`should render correctly with 2 mistakes`, () => {
    const wrapper = shallow(<GameMistakes mistakes={2} />);
    expect(wrapper).toMatchSnapshot();
  });
});
