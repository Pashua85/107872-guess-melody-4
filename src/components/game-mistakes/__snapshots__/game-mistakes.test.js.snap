// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameMistkes should render correctly with 2 mistakes 1`] = `ShallowWrapper {}`;

exports[`GameMistkes should render correctly with no mistakes 1`] = `ShallowWrapper {}`;
